app/src/main/java/net/squanchy/support/widget/ImageViewWithForeground.kt

```diff
@@ -0,0 +1,193 @@
+/*
+ * Portions derived from https://gist.github.com/chrisbanes/9091754
+ * For those portions:
+ *
+ * Copyright (C) 2006 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.squanchy.support.widget
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Rect
+import android.graphics.drawable.Drawable
+import android.support.v7.widget.AppCompatImageView
+import android.util.AttributeSet
+import android.view.Gravity
+import net.squanchy.R
+
+open class ImageViewWithForeground @JvmOverloads constructor(
+        context: Context,
+        attrs: AttributeSet,
+        defStyleAttr: Int = 0
+) : AppCompatImageView(context, attrs, defStyleAttr), ViewWithForeground {
+
+    private var foregroundDrawable: Drawable? = null
+    private var gravity = Gravity.FILL
+
+    private var foregroundInPadding = true
+    private var foregroundBoundsChanged: Boolean = false
+
+    private val bounds = Rect()
+    private val overlayBounds = Rect()
+
+    init {
+        val a = context.obtainStyledAttributes(attrs, R.styleable.ImageViewWithForeground, defStyleAttr, 0)
+
+        foregroundGravity = a.getInt(R.styleable.ImageViewWithForeground_android_foregroundGravity, gravity)
+
+        val d = a.getDrawable(R.styleable.ImageViewWithForeground_android_foreground)
+        if (d != null) {
+            foreground = d
+        }
+
+        foregroundInPadding = a.getBoolean(R.styleable.ImageViewWithForeground_foregroundInsidePadding, true)
+
+        a.recycle()
+    }
+
+    override fun foregroundGravity(): Int = gravity
+
+    override fun setForegroundGravity(foregroundGravity: Int) {
+        if (this.gravity == foregroundGravity) {
+            return
+        }
+
+        var adjustedGravity = foregroundGravity
+        if (adjustedGravity and Gravity.RELATIVE_HORIZONTAL_GRAVITY_MASK == 0) {
+            adjustedGravity = adjustedGravity or Gravity.START
+        }
+
+        if (adjustedGravity and Gravity.VERTICAL_GRAVITY_MASK == 0) {
+            adjustedGravity = adjustedGravity or Gravity.TOP
+        }
+
+        this.gravity = adjustedGravity
+
+        if (this.gravity == Gravity.FILL && foregroundDrawable != null) {
+            val padding = Rect()
+            foregroundDrawable!!.getPadding(padding)
+        }
+
+        requestLayout()
+    }
+
+    // We actually want the instance comparison here
```

Comment:
Was this a comment to a suppression that has disappeared? If so we should delete it

Reply:
it's about explaining the `===`

Reply:
Yes but it was [explaining a `SuppressWarnings`](https://github.com/squanchy-dev/squanchy-android/pull/435/files/a13f4b7f166036dd62d237c8d296106175e04308#diff-211ad138a368ab7fbd3b6bfbd3d40dffL107), I would take it out or move it closer to the code it refers to